src/Model/Post.php

```diff
@@ -342,13 +342,14 @@ private static function selectViewForUser(string $view, $uid, array $selected =
 	 * @param array   $selected  Array of selected fields, empty for all
 	 * @param array   $condition Array of fields for condition
 	 * @param array   $params    Array of several parameters
+	 * @param bool    $user_mode true = post-user-view, false = post-view
 	 *
 	 * @return boolean|object
 	 * @throws \Exception
 	 */
-	public static function selectForUser($uid, array $selected = [], array $condition = [], $params = [])
+	public static function selectForUser($uid, array $selected = [], array $condition = [], $params = [], bool $user_mode = true)
```

Comment:
Please create another method instead of a boolean flag parameter to better explain the difference between using either tables.

Reply:
I just added the parameter that I had added to the other similar functions as well.

Reply:
Concerning a different method name: The best way would be to move all `Model/Post` methods to `Model/Post/User`. But this is a slightly bigger task.

Reply:
I hear you, but until this is done, a better way would be to create another method with the different table name instead of a boolean flag parameter.

Reply:
Like said, this (ugly) parameter is used all around that class:
- https://github.com/friendica/friendica/blob/2021.06-rc/src/Model/Post.php#L135
- https://github.com/friendica/friendica/blob/2021.06-rc/src/Model/Post.php#L156
- https://github.com/friendica/friendica/blob/2021.06-rc/src/Model/Post.php#L172
- https://github.com/friendica/friendica/blob/2021.06-rc/src/Model/Post.php#L276

I would like to make a good solution - and this ugly parameter is a nice motivation to do it in the next release.

Also I don't know a nice function name, I must confess.

Reply:
Well, what's the functional difference between the `post-view` and the `post-user-view`?

Reply:
The `post-view` works with the `post` table as underlying structure, hence the `select` and other methods in the `Post` model class should work with that view. The `post-user-view` is working with the `post-user` table as underlying structure, so the functionality should move to the `Post\User` class.

Reply:
The problem here is: The `Post` class was meant as a drop in replacement for the `Item` class. And the `post-user-view` is the replacement for the `Item` table. Later I realized that queries on the `post` table could make sense under certain conditions (when we don't need to check for user permissions for example).

Reply:
I started reviewing the methods with the `$user_mode` parameter, and so far I'm confused. `Post::exists` is never called with the `$user_mode` parameter set, so it's always true. `Post::count` is only called with the `$user_mode = false` in `Factory/Api/Mastodon/Status->createFromUriId`, and when it is called with the default `$user_mode = true`, the condition doesn't always have a `uid`, which makes me wonder against what user are we checking the permissions?

Reply:
Do you mean this one? https://github.com/friendica/friendica/blob/2021.06-rc/mod/photos.php#L1279
There the query refers to the `parent` which is the `id` of the parent post in the `post-user` table. So we don't need the `uid` there.